sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -1431,4 +1431,28 @@ class AdaptiveQueryExecSuite
       }
     }
   }
+
+  test("SPARK-33933: AQE broadcast should not timeout with slow map tasks") {
```

Comment:
It seems that this case fails a little frequently：

- [https://amplab.cs.berkeley.edu/jenkins/job/spark-branch-3.1-test-maven-hadoop-2.7-jdk-11/95/](https://amplab.cs.berkeley.edu/jenkins/job/spark-branch-3.1-test-maven-hadoop-2.7-jdk-11/95/)
- [https://amplab.cs.berkeley.edu/jenkins/job/spark-master-test-maven-hadoop-3.2/1854/](https://amplab.cs.berkeley.edu/jenkins/job/spark-master-test-maven-hadoop-3.2/1854/)
- [https://amplab.cs.berkeley.edu/jenkins/job/spark-master-test-maven-hadoop-2.7-jdk-11-scala-2.13/147/#showFailuresLink](https://amplab.cs.berkeley.edu/jenkins/job/spark-master-test-maven-hadoop-2.7-jdk-11-scala-2.13/147/#showFailuresLink)
- [https://amplab.cs.berkeley.edu/jenkins/job/spark-branch-3.1-test-maven-hadoop-3.2/80/](https://amplab.cs.berkeley.edu/jenkins/job/spark-branch-3.1-test-maven-hadoop-3.2/80/)

@zhongyu09 @cloud-fan  @viirya

Reply:
Hmm, so it is still flaky.

Reply:
I try the test tens of times and the test failed twice. As we discussed, the guarantee is not strict since the submit of broadcast job and shuffle map job are in different thread. So there's still risk for the shuffle map job schedule earlier before broadcast job. I wonder should we need to remove the UT until we thorough resolve the issue.

CC @cloud-fan @dongjoon-hyun

Reply:
I run `SPARK-33933: AQE broadcast should not timeout with slow map tasks`  in local 5 times and all failed as follow:

```
- SPARK-33933: AQE broadcast should not timeout with slow map tasks *** FAILED ***
  1751 was not greater than 2000 (AdaptiveQueryExecSuite.scala:1454)
```

Reply:
> Hmm, so it is still flaky.

yes

Reply:
For now, revert the PR or remove the UT, which one is better?

Reply:
I was worried about this and tested the PR twice...

Maybe it's better to write a UT to verify the stage submission order, which is more reliable. We can use SparkListener to watch the job start events, and see if the job with less tasks (the broadcast stage) is started first.

@zhongyu09 what do you think?

Reply:
I am afraid that won't work. The real materialize work runs in different thread for broadcast and map tasks. So the job sequence of start events might not be strictly the same as calling materialize().

Reply:
we can add retry logic to make it less likely to fail. It should be much more stable than checking the running time.

Another way is to check the stage submission order (not the job submission order). But seems there is no easy way to get the stage submission order. Maybe we can add logs and check the log?

Reply:
@cloud-fan  You mean the error 1751 was not greater than 2000 (AdaptiveQueryExecSuite.scala:1454) ? That's my fault, by moving the UT from BroadcastJoinSuite to AdaptiveQueryExecSuite, the spark conf changed to local[2] and so the running times are faster than before.

Reply:
> the spark conf changed to local[2] and so the running times are faster than before.

This shows the test is unreliable...

Checking the Spark jobs submission order should be easy to do and fast to run, and with retry it should be unlikely to fail. It's better to check stage submission order directly, if we can figure out how to do it.

Reply:
Yes I know, the failure reported by @LuciferYang is easy to solve. 
But the question is, the jobs submission order may be not correct, like the test in https://github.com/apache/spark/pull/31084.

Reply:
@cloud-fan  You mean `with retry it should be unlikely to fail` to solve the edge case?

Reply:
I am trying to create a new PR. Sorry for the inconvenience.

Reply:
We can get the stage submission time using SparkListeneer. But after trying serval times, the stage submission time is not stable thus the UT cannot always passed. I suggest to remove the UT before we completely solve the issue. https://github.com/apache/spark/pull/31099